src/Hosting/Hosting/src/WebHostBuilder.cs

```diff
@@ -269,7 +269,8 @@ private IServiceCollection BuildCommonServices(out AggregateException hostingSta
             _configureAppConfigurationBuilder?.Invoke(_context, builder);
 
             var configuration = builder.Build();
-            services.AddSingleton<IConfiguration>(configuration);
+            // register configuration as factory to make it dispose with the service provider
+            services.AddSingleton<IConfiguration>(_ => configuration);
```

Comment:
Also needs to set shouldDisposeConfiguration https://github.com/aspnet/AspNetCore/blob/b8317c9f768460b509e95925a65d3ab78feba779/src/Hosting/Hosting/src/WebHostBuilder.cs#L267

Reply:
Oh, right, cross repo change..

Reply:
Good that you mention it though. I probably would have forgotten it after all.

For the repo changes, I have to wait until the maestro bot detects the changes and makes a PR, right?

Reply:
I honestly don’t know how that would work.. I’ve looked at the file that contains all the package version numbers but it says that it should only be updated automatically using “Darc”. But the Darc readme basically tells me that I won’t be able to use it… 😅 

Guess I’ll just wait for the dependencies to be updated 🤷‍♂️

Reply:
Sounds like you're in the **darc** about dependency updates.
<br />
<br />
<br />
<br />
<br />
<br />
<br />
<br />
... I'll show myself out.